### ISSUE TITLE
Configured Queues are not selectable

### ISSUE THREAD

**Issue by gkieninger** [django-commons/django-tasks-scheduler#152]
**Describe the bug**
I can't select my defined QUEUES in the repeatabletasks and other tasks.
I can see my defined Queues only in the admin of the queues.

**To Reproduce**
Define SCHEDULER_QUEUES like this

SCHEDULER_QUEUES = {
    'default': {
        'HOST': 'redis',
        'PORT': 6379,
        'DB': 0,
    },
    'test': {
        'HOST': 'redis',
        'PORT': 6379,
        'DB': 0,
    },
}

go to ADMIN to the repeatabletask
add a new task
try to select the queue 'test', but you only see 'default', 'low' and 'high'

**Expected behavior**
I would like to see my defined QUEUES in the select of the tasks. So I can only use the predefined queues.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: macos
 - python version: 3.11
 - django version: 4
 - requirements.txt?

**Additional context**

**Comment by cunla** [comment]
Which version of django-tasks-schedule are you using?

This is related to an issue with django 4.x not supporting variable choices (https://github.com/django-commons/django-tasks-scheduler/issues/132).

Unfortunately, I couldn't find a good way to solve it for django 4.x.

**Comment by gkieninger** [comment]
I upgraded to the newest version with Django 5.1 and now it works like expected. I will give this a try and hope everything is fine.

**Comment by DhavalGojiya** [comment]
@cunla | @gkieninger 

The issue is not related to the Django version. The problem lies with `django-task-scheduler==1.3.3` and `django-task-scheduler==1.3.4`, where the model choice argument is removed and no longer takes user-defined `RQ_QUEUES` or `SCHEDULER_QUEUES`. Instead, it has been hardcoded as follows:

### Code in v1.3.3 & v1.3.4 (Location: `scheduler/models/scheduled_task.py`):

```python
class BaseTask(models.Model):
    created = models.DateTimeField(auto_now_add=True)
    modified = models.DateTimeField(auto_now=True)
    QUEUES = [("default", "default"), ("low", "low"), ("high", "high")]
    # Some other fields
    
    queue = models.CharField(
        _('queue'), max_length=255, choices=QUEUES,
        help_text=_('Queue name'), )
```

This is the issue, and it affects only versions `v1.3.3` and `v1.3.4`. You can find the relevant code here:
- [Link to v1.3.3](https://github.com/django-commons/django-tasks-scheduler/blob/v1.3.3/scheduler/models/scheduled_task.py#L66)
- [Link to v1.3.4](https://github.com/django-commons/django-tasks-scheduler/blob/v1.3.4/scheduler/models/scheduled_task.py#L66)

I'm currently using `v1.3.3` and fixed this dropdown choice by replacing the `QUEUES` variable as follows:

### Working Code for v1.3.3 & v1.3.4:

```python
class BaseTask(models.Model):
    created = models.DateTimeField(auto_now_add=True)
    modified = models.DateTimeField(auto_now=True)
    QUEUES = [(key, key) for key in settings.QUEUES.keys()]
    # Some other fields
    
    queue = models.CharField(
        _('queue'), max_length=255, choices=QUEUES,
        help_text=_('Queue name'), )
```

I have already forked the repo with these changes. Should I raise a PR for this fix? @cunla 

Also, could you suggest the appropriate target branch in your official repository for my bugfix branch?

**Comment by cunla** [comment]
Hi,

Once you change the `QUEUES = [(key, key) for key in settings.QUEUES.keys()]`, every time you run `makemigrations` it will create a new migration for your specific queues, and therefore receiving further migrations from the package would result in an error since there are 2 paths.

**Comment by DhavalGojiya** [comment]
> Hi,
> 
> Once you change the `QUEUES = [(key, key) for key in settings.QUEUES.keys()]`, every time you run `makemigrations` it will create a new migration for your specific queues, and therefore receiving further migrations from the package would result in an error since there are 2 paths.

So true. Do you have any fix for this now? People using v1.3.3 and v1.3.4 cannot select the queue dropdown for their own settings-defined queue.

**Comment by cunla** [comment]
Considering there are not going to be new features in v1.3.4 - There should not be new migrations created for these versions as well. Therefore, I believe it is safe to allow creating this new migration.

However, I really recommend upgrading to django 5 and django-tasks-scheduler v2+
